docs/sources/setup-grafana/configure-docker.md

```diff
@@ -57,48 +91,217 @@ The following settings are hard-coded when launching the Grafana Docker containe
 
 Example:
 
+The following example runs Grafana using the `console file` log mode that is set in the `GF_LOG_MODE` environment variable.
+
 ```bash
-# Run Grafana while logging to both standard out and /var/log/grafana/grafana.log
+# Run Grafana while logging to both standard out
+# and /var/log/grafana/grafana.log
+
 docker run -p 3000:3000 -e "GF_LOG_MODE=console file" grafana/grafana-enterprise
 ```
 
 ## Configure Grafana with Docker Secrets
 
-> Only available in Grafana v5.2 and later.
+Using configuration files, you can input confidential data like login credentials and secrets into Grafana. This method works well with [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/), as the secrets are automatically mapped to the `/run/secrets/` location within the container.
 
-It's possible to supply Grafana with configuration through files. This works well with [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/) as the secrets by default gets mapped into `/run/secrets/<name of secret>` of the container.
+You can apply this technique to any configuration options in `conf/grafana.ini` by setting `GF_<SectionName>_<KeyName>__FILE` to the file path that contains the secret information. For more information about Docker secret command usage, refer to [docker secret](https://docs.docker.com/engine/reference/commandline/secret/).
 
-You can do this with any of the configuration options in conf/grafana.ini by setting `GF_<SectionName>_<KeyName>__FILE` to the path of the file holding the secret.
-
-For example, you could set the admin password this way:
+The following example demonstrates how to set the admin password:
 
 - Admin password secret: `/run/secrets/admin_password`
 - Environment variable: `GF_SECURITY_ADMIN_PASSWORD__FILE=/run/secrets/admin_password`
 
-## Configure AWS credentials for CloudWatch Support
+### Configure Docker secrets credentials for AWS CloudWatch
+
+Grafana ships with built-in support for the [Amazon CloudWatch datasource]({{< relref "../datasources/aws-cloudwatch" >}}). To configure the data source, you must provide information such as the AWS ID-Key, secret access key, region, and so on. You can use Docker secrets as a way to provide this information.
+
+Example:
+
+The following example defines how to use Grafana environment variables via Docker Secrets for the AWS ID-Key, secret access key, region, and profile.
 
 ```bash
-docker run -d \
--p 3000:3000 \
---name=grafana \
--e "GF_AWS_PROFILES=default" \
--e "GF_AWS_default_ACCESS_KEY_ID=YOUR_ACCESS_KEY" \
--e "GF_AWS_default_SECRET_ACCESS_KEY=YOUR_SECRET_KEY" \
--e "GF_AWS_default_REGION=us-east-1" \
+docker run -d
+-p 3000:3000
+--name=grafana
+-e "GF_AWS_PROFILES=default"
+-e "GF_AWS_default_ACCESS_KEY_ID=YOUR_ACCESS_KEY"
+-e "GF_AWS_default_SECRET_ACCESS_KEY=YOUR_SECRET_KEY"
```

Comment:
It may make sense to show how this would be done using a docker secret, rather than encouraging people to pass it via the command line.